### PR TITLE
Batch changing and pagination for API credentials is not working

### LINKED ISSUE #11860: Batch changing and pagination for API credentials is not working
Batch change options are missing (let's write a test for that?)

[screenshot-8080-escopecz-mautic-dhvhavfl3xd ws-eu81 gitpod io-2023 01 10-11_40_56](https://user-images.githubusercontent.com/2930593/211543299-b8e6b498-b274-40b3-ac54-9077f65f812d.png)

Pagination is not working: https://watch.screencastify.com/v/KD7RJQfWQ7A2lsVjmxCJ (can we do a test for that as well?)

_Originally posted by @RCheesley in https://github.com/mautic/mautic/pull/11853#pullrequestreview-1242049885_
      

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://app.bountysource.com/issues/121828510-batch-changing-and-pagination-for-api-credentials-is-not-working?utm_campaign=plugin&utm_content=tracker%2F5355074&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://app.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F5355074&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by shinde-rahul**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴
| New feature/enhancement? (use the a.x branch)      | 🔴
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #11860 

## Description

---
#### For Pager
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create Credentials, minimum 6
3. Select the pager to limit 5
4. Check the Pager is working.

#### For Click all
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create Credentials
3. There will be no check all box.

### CODE REVIEW

**Comment by RCheesley** [comment]
@shinde-rahul can you check the failing test please? Thanks!

**Comment by shinde-rahul** [comment]
@escopecz, I don't know why the update controller is failing. I would need some guidance to fix this.

**Comment by escopecz** [comment]
@shinde-rahul can you be more concrete please? What update controller? Is there any error?

**Comment by shinde-rahul** [comment]
@escopecz, The code changes are not related to failing the test.

**Comment by escopecz** [comment]
@shinde-rahul I restarted the tests several times. It looks like it is related to these code changes. Are you able to replicate the problem when you run the test or the whole test suite locally?

**Comment by shinde-rahul** [comment]
@escopecz, Local runs without failure,
```
OK, but incomplete, skipped, or risky tests!
Tests: 4123, Assertions: 13229, Skipped: 5.
```

I ran this multiple times with the above output.

**Comment by escopecz** [comment]
Having a problem is one thing, but being unable to reproduce it sucks.

In these cases I try:
1. Use the same config file as the CI does.
2. If that won't help, try to comment out the test that you've added and let the CI run to see if the problem is in the test. It can set some constant that the following test needs or it can corrupt the DB connection in some way. This way you'll know the problem is the test.
3. If that's not the case, try to revert the code changes to see if that's it.

File: app/bundles/ApiBundle/Tests/Functional/Controller/ClientControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class ClientControllerTest extends MauticMysqlTestCase
+{
+    private const TOTAL_COUNT = 20;
+
+    public function testIndexActionForPager(): void
+    {
+        $this->createApiClients();
```
Comment (shinde-rahul) [inline app/bundles/ApiBundle/Tests/Functional/Controller/ClientControllerTest.php:21]:
```suggestion
        $this->markTestSkipped('Disabling to verify the cause of failure');
        $this->createApiClients();
```

**Review by escopecz — APPROVED** [review]
I can see that this PR is fixing the pagination and sorting of API credentials by name. It also removes the "check all" checkbox. There are still checkboxes on each row that do nothing so it may be worth to hide them too. But this is still a great improvement, so I'm OK to merge this as is 👍 
Screenshot 2024-10-15 at 10 14 29

**Comment by andersonjeccel** [comment]
@shinde-rahul I'd gently ask you to remove all checkboxes. Would you be up for that?

**Comment by shinde-rahul** [comment]
@andersonjeccel, I am okay doing it, and hoping to get done in another PR for making the Developer experience better. There are few logical issues with table header for all listings.

Whats your thoughts?

**Comment by andersonjeccel** [comment]
@shinde-rahul wondering if things would work if we just delete the `<span class="input-group-addon"></span>` _(I'm not aware of how they're built currently, is it possible?)_

**Comment by shinde-rahul** [comment]
@andersonjeccel, We need to think about this view,
Screenshot 2024-10-15 at 8 02 44 PM

After removing the checkbox, this does not look good, and the vertical three dots need to make more sense.

This is why I propose doing this in a separate PR.

**Comment by andersonjeccel** [comment]
Got it, so leave it this for another PR

**Comment by abhisekmazumdar** [comment]
The code changes make sense to me.
also I see the pagination working after applying these changes:
image

**Pull request merged**